streams/src/main/java/org/apache/kafka/streams/kstream/Predicate.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream;
+
+public interface Predicate<K, V> {
+
+    boolean apply(K key, V value);
```

Comment:
If the `apply` method were changed to `test`, then when Kafka moves to Java 8 this interface could extend `java.util.function.BiPredicate<K,V>`. That wouldn't really benefit anyone that simply supplied their own lambdas, but it might make it a bit easier or more obvious that existing `BiPredicate` implementations could be passed in. Regardless, aligning with Java 8 might be a good thing in and of itself.